src/libraries/System.DirectoryServices.Protocols/src/System/DirectoryServices/Protocols/ldap/LdapConnection.Linux.cs

```diff
@@ -35,5 +36,44 @@ private int InternalBind(NetworkCredential tempCredential, SEC_WINNT_AUTH_IDENTI
 
             return error;
         }
+
+        private int BindSasl()
+        {
+            SaslDefaultCredentials defaults = GetSaslDefaults();
+            IntPtr ptrToDefaults = Marshal.AllocHGlobal(Marshal.SizeOf(defaults));
```

Comment:
how big usually the size of defaults? I am just wondering if we need to use the stackalloc for small sizes to avoid the allocations?

Reply:
not big at all, it is just one of these

```c#
    [StructLayout(LayoutKind.Sequential)]
    internal struct SaslDefaultCredentials
    {
        public string mech;
        public string realm;
        public string authcid;
        public string passwd;
        public string authzid;
    }
```

which will usually only have mech and authcid set, with all the other strings set to NULL.